src/main/java/com/payline/payment/cvconnect/utils/http/HttpClient.java

```diff
@@ -0,0 +1,261 @@
+import com.payline.pmapi.bean.common.FailureCause;
+import com.payline.pmapi.logger.LogManager;
+import org.apache.http.Header;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicHeader;
+import org.apache.logging.log4j.Logger;
+
+import javax.net.ssl.HttpsURLConnection;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+
+
+public class HttpClient {
+
+    private static final Logger LOGGER = LogManager.getLogger(HttpClient.class);
+
+    //Headers
+    private static final String CONTENT_TYPE_KEY = "Content-Type";
+    private static final String CONTENT_TYPE_VALUE = "application/json";
+    private static final String ANCV_SECURITY = "ANCV-Security";
+
+    // Paths
+    private static final String BASE_PATH = "payment-transactions";
+    private static final String CONFIRM_TRANSACTION_PATH = "payer";
+    private static final String CANCELLATION_PATH = "cancellation";
+
+    // Exceptions messages
+    private static final String SERVICE_URL_ERROR = "Service URL is invalid";
+
+    /**
+     * The number of time the client must retry to send the request if it doesn't obtain a response.
+     */
+    private int retries;
+
+    private org.apache.http.client.HttpClient client;
+
+    // --- Singleton Holder pattern + initialization BEGIN
+
+    /**
+     * ------------------------------------------------------------------------------------------------------------------
+     */
+    private HttpClient() {
+        int connectionRequestTimeout;
+        int connectTimeout;
+        int socketTimeout;
+        try {
+            // request config timeouts (in seconds)
+            ConfigProperties config = ConfigProperties.getInstance();
+            connectionRequestTimeout = Integer.parseInt(config.get("http.connectionRequestTimeout"));
+            connectTimeout = Integer.parseInt(config.get("http.connectTimeout"));
+            socketTimeout = Integer.parseInt(config.get("http.socketTimeout"));
+
+            // retries
+            this.retries = Integer.parseInt(config.get("http.retries"));
+        } catch (NumberFormatException e) {
+            throw new PluginException("plugin error: http.* properties must be integers", e);
+        }
+
+        RequestConfig requestConfig = RequestConfig.custom()
+                .setConnectionRequestTimeout(connectionRequestTimeout * 1000)
+                .setConnectTimeout(connectTimeout * 1000)
+                .setSocketTimeout(socketTimeout * 1000)
```

Comment:
de manière générale, les configuration de EndPoint, timeout...
sont configurées via les PARTNER_CONFIGURATION en Base de données et donc récupérées 

Ces données :
# --- HTTP Client config ---
# the time to wait for a connection from the connection manager/pool (seconds)
http.connectionRequestTimeout=5
# the time to establish the connection with the remote host (seconds)
http.connectTimeout=5
# the time waiting for data – after establishing the connection (seconds)
http.socketTimeout=10
# the number of times the client should retry to reach the remote host
http.retries=3

Reply:
Je ne suis pas contre l'idée de mettre ces parametres dans les PartnerConfig mais il faut d'abord tracer ca dans les bonnes pratiques: https://docs.monext.fr/pages/viewpage.action?pageId=748248838